components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -1220,26 +1220,13 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
           ! The rates are adjusted here (where necessary) such that the sum of the sinks of mass cannot
           ! be greater than the sum of the sources, thereby resulting in overdepletion.
 
-          !-- Limit ice process rates to prevent overdepletion of sources such that
-          !   the subsequent adjustments are done with maximum possible rates for the
-          !   time step.  (note: most ice rates are adjusted here since they must be done
-          !   simultaneously (outside of iice-loops) to distribute reduction proportionally
-          !   amongst categories.
-          !PMC - might need to rethink above statement since only one category now.
-          ! AaronDonahue: Do we need the below checks for the new definition of
-          ! how qidep and qisub are derived?
-          dumqvi = qv_sat(t(i,k),pres(i,k),1)
-          qdep_satadj = (qv(i,k)-dumqvi)/(1._rtype+xxls(i,k)**2*dumqvi/(cp*rv*t(i,k)**2))*odt
-          qidep  = qidep*min(1._rtype,max(0._rtype, qdep_satadj)/max(qidep, 1.e-20_rtype))
-          qisub  = qisub*min(1._rtype,max(0._rtype,-qdep_satadj)/max(qisub, 1.e-20_rtype))
-          !==
-
           ! vapor -- not needed, since all sinks already have limits imposed and the sum, therefore,
           !          cannot possibly overdeplete qv
 
           ! cloud
           sinks   = (qcaut+qcacc+qccol+qcevp+qcheti+qcshd+qiberg)*dt
           sources = qc(i,k) + (qccon+qcnuc)*dt
+          ratio = 0.0 ! Initialize ratio in order to only limit qidep and qisub when needed.  See below.
```

Comment:
I still don't see how ratio = 0.0 is ever used. Won't it be set 100% of the time in the if/else below? Also, shouldn't it be _rtype if we did keep it?

Reply:
Ahh yes good point, this is actually an incorrect fix.  The idea was that if there is qc but no sinks we don't want the next if statement to remove all qidep and qisub, in other words we want ratio to be zero.  But you are right, as it is currently set up ratio will always be set in sinks vs. sources if statement.  I think the correct fix is to add an elseif for the special case where qidep and qisub shouldn't be limited.  I'll make the change.